tools/calc_integrity.py

```diff
@@ -14,19 +14,27 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import validators
+import json
 import os
+import pathlib
 import sys
 
-from registry import read
-from registry import download
-from registry import integrity
+import validators
+from registry import download, integrity, read
 
 if __name__ == "__main__":
-  # Under 'bazel run' we want to run within the source folder instead of the execroot.
-  if os.getenv("BUILD_WORKSPACE_DIRECTORY"):
-    os.chdir(os.getenv("BUILD_WORKSPACE_DIRECTORY"))
-  if validators.url(sys.argv[1]):
-    print(integrity(download(sys.argv[1])))
-  else:
-    print(integrity(read(sys.argv[1])))
+    # Under 'bazel run' we want to run within the source folder instead of the execroot.
+    if os.getenv("BUILD_WORKSPACE_DIRECTORY"):
+        os.chdir(os.getenv("BUILD_WORKSPACE_DIRECTORY"))
+    target = sys.argv[1]
+    if validators.url(target):
+        print(integrity(download(target)))
+    elif (file := pathlib.Path(target)).is_file() and file.name == "source.json":
```

Comment:
I really like the feature, I just find this condition somewhat adhoc. How about adding a separate `update_integrity.py` script for this purpose? It could also use more concise arguments, such as a module name and a version instead of the full path.

Reply:
I like the concise arguments idea.?